pkg/types/types.go

```diff
@@ -261,10 +265,28 @@ func MergeMetrics(metrics [][]Metric, replicaMismatchConfig cfg.RenderReplicaMis
 			zap.Int("replica_points_total", metricsStat.DataPointCount),
 		)
 	}
+	if len(diffResolutionMetrics) > 0 {
+		logger.Info("metric with different schema observed",
+			zap.Strings("schema_mismatch_metrics", diffResolutionMetrics),
+			zap.Int("metrics total", len(diffResolutionMetrics)),
+		)
+	}
 
 	return merged, metricsStat
 }
 
+func isStepTimeMatching(ms []Metric) bool {
```

Comment:
Have we checked the performance impact of this? Is the performance impact justified to have this information?